lib/elastic_apm.rb

```diff
@@ -381,5 +391,29 @@ def add_filter(key, callback = nil, &block)
 
       agent&.add_filter(key, block || callback)
     end
+
+    def set_transaction_outcome(result: nil, http_status_code: nil)
+      return unless transaction = current_transaction
+      return transaction.outcome = result if result
+      return unless http_status_code
```

Comment:
I think this is a bit hard to read. I think a few extra newlines or a comment would help us in the future.

If both `result` and `http_status_code` are `nil`, why even call the method? Is that instead an ArgumentError? What if both are passed?

I think I feel the API of the method is a bit vague.

Reply:
Also, the knowledge of HTTP status codes might not tie directly to the concept of _outcome_, hence it might not even belong here, but in the surrounding context ie. the middleware.